### PR TITLE
Add decorator class and it's sub classes

### PR DESCRIPTION
**Pull request by ChrispaixK**
# OOP school library App 📚 
> Add decorator class and its subclasses

In this PR I :
* Create `Nameable` class
* Add inheritance from `Nameable` on `Person` class
* Create `Decorator` class
* Create `CapitalizeDecorator` and `TrimmerDecorator` classes that inherit from `Decorator` class
* Fix linters errors

### CODE REVIEW

File: decorator.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+require './nameable'
+
+class Decorator
+  def initialize(nameable)
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable
+  end
```
Comment (dasileker) [inline decorator.rb]: - [x] Already initialized in the constructor it should be called @nameable.correct_name

File: capitalize.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+require './decorator'
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.capitalize
```
Comment (dasileker) [inline capitalize.rb]: - [x] Also it should be corrected before capitalization. hint: `@neamable.correct_name.capaitalize`

File: trimmer.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+require_relative './decorator'
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    @nameable[0, 10]
```
Comment (dasileker) [inline trimmer.rb]: - [x] The should have a maximum of 10 characters. If it's longer it should trim the word.
- [x]  The code is a decorator that will correct the name of an object.
 The code first checks to see if the length of the string is greater than 10, and then it returns whatever number is in between 0 and 9.
 The code will return the correct name of the person's first and last names.
 The code will return a string with no spaces in it

**Review by dasileker — CHANGES_REQUESTED** [review]
### **Hi @ChrispaixK,** 

### **Good job so far!**
There are some issues that you still need to work on to go to the next project but you are almost there!

## **Highlights**
- The PR is very PRO👌.
- The Codes are clean 👌.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: trimmer.rb (hunk 3 above)
Comment (ChrispaixK) [reply trimmer.rb]: thanks for the clarifications @dasileker, I get it 👍🏾  😄

**Review by julie-ify — APPROVED** [review]
Hi @ChrispaixK,

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**